rascal-lsp/src/main/rascal/library/demo/lang/pico/LanguageServer.rsc

```diff
@@ -61,9 +71,46 @@ set[LanguageService] picoLanguageServer(bool allowRecovery) = {
     codeAction(picoCodeActionService),
     rename(picoRenamingService, prepareRenameService = picoRenamePreparingService),
     didRenameFiles(picoFileRenameService),
-    selectionRange(picoSelectionRangeService)
+    selectionRange(picoSelectionRangeService),
+    formatting(picoFormattingService)
 };
 
+list[TextEdit] picoFormattingService(Focus input, FormattingOptions opts) {
+    str original = "<input[-1]>";
+    box = toBox(input[-1]);
+    box = visit (box) { case i:I(_) => i[is=opts.tabSize] }
```

Comment:
This is not a good example for users on how to parameterize the indentation level. The reason is that not all `I` boxes will correspond to the default indentation level in a typical language. Some are different. 

So indentation level should be a parameter to `toBox` where the spec writer can decide where to use it and where not.

Reply:
It's good to discover this now and here. A few minor additions to `toBox` should help. We have to think hard how to make passing the parameters simple and easy, because toBox is highly recursive and we might skip accidentally to pass it on..